samcli/commands/deploy/deploy_context.py

```diff
@@ -174,6 +177,42 @@ def deploy(
         fail_on_empty_changeset=True,
         confirm_changeset=False,
     ):
+        """
+        Deploy the stack to cloudformation.
+        - if changeset needs confirmation, it will prompt for customers to confirm.
+        - if no_execute_changeset is True, the changeset won't be executed.
+
+        Parameters
+        ----------
+        stack_name : str
+            name of the stack
+        template_str : str
+            the string content of the template
+        parameters : List[Dict]
+            List of parameters
+        capabilities : List[str]
+            List of capabilities
+        no_execute_changeset : bool
+            A bool indicating whether to execute changeset
+        role_arn : str
+            the Arn of the role to create changeset
+        notification_arns : List[str]
+            Arns for sending notifications
+        s3_uploader : S3Uploader
+            S3Uploader object to upload files to S3 buckets
+        tags : List[str]
+            List of tags passed to CloudFormation
+        region : str
+            AWS region to deploy the stack to
+        fail_on_empty_changeset : bool
+            Should fail when changeset is empty
+        confirm_changeset : bool
+            Should wait for customer's confirm before executing the changeset
+
+        Returns
```

Comment:
Not sure this is necessary if the method doesn't return anything (VSCode doesn't add it)

Reply:
Good question. I don't find standard about this one in the official doc. https://numpydoc.readthedocs.io/en/latest/format.html

In PyCharm, Returns is added even it returns `None`. And this stackoverflow post suggests to leave a None here with 42 upvotes

https://stackoverflow.com/questions/27784179/docstrings-when-nothing-is-returned

> You should use None, as that is what your function actually returns:

Reply:
update, I think we should remove it, I looked into the numpy source code and numpy itself does not add "Returns None" 

https://github.com/numpy/numpy/blob/5cae51e794d69dd553104099305e9f92db237c53/numpy/lib/format.py#L455-L468

```py
def write_array_header_2_0(fp, d):
    """ Write the header for an array using the 2.0 format.
        The 2.0 format allows storing very large structured arrays.
    .. versionadded:: 1.9.0
    Parameters
    ----------
    fp : filelike object
    d : dict
        This has the appropriate entries for writing its string
        representation to the header of the file.
    """
    _write_array_header(fp, d, (2, 0))
```

Reply:
There is actually a pylint rule for this "redundant-returns-doc"

All fixed